renderlib/command.cpp

```diff
@@ -535,57 +542,59 @@ SetTimeCommand::execute(ExecutionContext* c)
     return;
   }
 
-  struct STAT64_STRUCT buf;
-  if (STAT64_FUNCTION(c->m_currentFilePath.c_str(), &buf) == 0) {
-    VolumeDimensions dims;
-    // note T and S args are swapped in order here. this is intentional.
-    std::shared_ptr<ImageXYZC> image =
-      FileReader::loadFromFile(c->m_currentFilePath, &dims, m_data.m_time, c->m_currentScene);
-    if (!image) {
-      return;
-    }
-
-    c->m_appScene->m_timeLine.setCurrentTime(m_data.m_time);
+  LoadSpec loadSpec = c->m_loadSpec;
+  loadSpec.time = m_data.m_time;
+  std::shared_ptr<ImageXYZC> image;
+  try {
+    image = FileReader::loadFromFile(loadSpec);
+  } catch (...) {
+    LOG_ERROR << "Failed to load time " << m_data.m_time << " from file " << c->m_loadSpec.toString();
+    image = nullptr;
+  }
+  if (!image) {
+    LOG_WARNING << "SetTime command called without a file loaded";
+    return;
+  }
 
-    // we expect the scene volume dimensions to be the same; we want to preserve all view settings here.
-    // BUT we want to convert the old lookup tables to new lookup tables
-    // if we are preserving absolute transfer function settings
+  // successfully loaded; update loadspec in context
+  c->m_loadSpec = loadSpec;
 
-    // assume sizeC is same for both previous image and new image!
-    if (image->sizeC() != c->m_appScene->m_volume->sizeC()) {
-      LOG_ERROR << "Channel count mismatch for different times in same file";
-    }
+  c->m_appScene->m_timeLine.setCurrentTime(m_data.m_time);
 
-    // remap LUTs to preserve absolute thresholding
-    for (uint32_t i = 0; i < image->sizeC(); ++i) {
-      GradientData& lutInfo = c->m_appScene->m_material.m_gradientData[i];
-      lutInfo.convert(c->m_appScene->m_volume->channel(i)->m_histogram, image->channel(i)->m_histogram);
+  // we expect the scene volume dimensions to be the same; we want to preserve all view settings here.
+  // BUT we want to convert the old lookup tables to new lookup tables
+  // if we are preserving absolute transfer function settings
 
-      image->channel(i)->generateFromGradientData(lutInfo);
-    }
+  // assume sizeC is same for both previous image and new image!
```

Comment:
if we are checking, are we still assuming?  Or am I misunderstanding something here?

Reply:
Maybe I'm misusing the word assume, and I'm making the assumption explicit in code - "it's an error if they are different" but I still want to log the error in this app's particular logging system so that if it ever comes up I'll know what happened.  (But this isn't new code, it's just in the diff because the if statement containing this was removed.)

Reply:
Maybe 'require sizeC to be the same for both previous image and new image'?